scvid/data/read.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from anndata import AnnData, read_h5ad
+from google.cloud import storage
+
+
+def read_h5ad_gcs(filename: str) -> AnnData:
+    r"""
+    Read `.h5ad`-formatted hdf5 file from the Google Cloud Storage.
+
+    Example::
+
+        >>> adata = read_h5ad_gcs("gs://dsp-cell-annotation-service/benchmark_v1/benchmark_v1.000.h5ad")
+
+    Args:
+        filename (str): Path to the data file in Cloud Storage.
+    """
+    assert filename.startswith("gs:")
+    # parse bucket and blob names from the filename
+    filename = re.sub(r"^gs://?", "", filename)
+    bucket_name, blob_name = filename.split("/", 1)
+
+    storage_client = storage.Client()
```

Comment:
Consider the following change:

```
from google.cloud.storage import Client

def read_h5ad_gcs(filename: str, storage_client: Optional[Client]):
    ...
    if storage_clientis None:
        storage_client= Client()
    ... 
```

We will soon find use-cases where the GCS storage client needs to be authenticated with privileged API tokens.